js/adapt-youtube.js

```diff
@@ -1,208 +1,206 @@
-      var widgetWidth = this.$('.component__widget').width();
+  setupEventListeners() {
+    this.completionEvent = (this.model.get('_setCompletionOn') || 'play');
 
-      $iframe.width(widgetWidth);
+    if (this.completionEvent !== 'inview') return;
+    this.setupInviewCompletion('.component__widget');
+  }
 
-      var aspectRatio = (this.model.get('_media')._aspectRatio ? parseFloat(this.model.get('_media')._aspectRatio) : 1.778);//default to 16:9 if not specified
-      if (!isNaN(aspectRatio)) {
-        $iframe.height(widgetWidth / aspectRatio);
-      }
-    },
-
-    postRender: function() {
-      //FOR HTML/HBS Paramenters: https://developers.google.com/youtube/player_parameters
-      if (Adapt.youTubeIframeAPIReady === true) {
-        this.onYouTubeIframeAPIReady();
-      } else {
-        Adapt.once('youTubeIframeAPIReady', this.onYouTubeIframeAPIReady, this);
+  onYouTubeIframeAPIReady() {
+    this.player = new YT.Player(this.$('iframe').get(0), {
+      events: {
+        'onStateChange': this.onPlayerStateChange,
+        'onReady': this.onPlayerReady
       }
-    },
+    });
 
-    remove: function() {
-      if (this.player !== null) {
-        this.player.destroy();
-      }
+    this.isPlaying = false;
 
-      ComponentView.prototype.remove.call(this);
-    },
+    this.setReadyStatus();
 
-    setupEventListeners: function() {
-      this.completionEvent = (this.model.get('_setCompletionOn') || 'play');
+    this.setupEventListeners();
 
-      if (this.completionEvent === 'inview') {
-        this.setupInviewCompletion('.component__widget');
-      }
-    },
+    this.setIFrameSize();
+  }
 
-    onYouTubeIframeAPIReady: function() {
-      this.player = new YT.Player(this.$('iframe').get(0), {
-        events: {
-          'onStateChange': this.onPlayerStateChange,
-          'onReady': this.onPlayerReady
-        }
-      });
+  onMediaStop(view) {
+    // if it was this view that triggered the media:stop event, ignore it
+    if (view && view.cid === this.cid) return;
 
-      this.isPlaying = false;
+    if (!this.isPlaying) return;
+    this.player.pauseVideo();
+  }
 
-      this.setReadyStatus();
+  onPlayerReady() {
+    if (!this.model.get('_media')._playbackQuality) return;
+    this.player.setPlaybackQuality(this.model.get('_media')._playbackQuality);
+  }
 
-      this.setupEventListeners();
+  /**
+  * this seems to have issues in Chrome if the user is logged into YouTube (possibly any Google account) - the API just doesn't broadcast the events
+  * but instead throws the error:
+  * Failed to execute 'postMessage' on 'DOMWindow': The target origin provided ('https://www.youtube.com') does not match the recipient window's origin ('http://www.youtube.com').
+  * This is documented here:
+  *   https://code.google.com/p/gdata-issues/issues/detail?id=5788
+  * but I haven't managed to get any of the workarounds to work... :-(
+  */
+  onPlayerStateChange(e) {
+    switch(e.data) {
```

Comment:
ESLint: needs space.
```suggestion
    switch (e.data) {
````